include/ddc/kernels/splines/matrix_banded.hpp

```diff
@@ -0,0 +1,138 @@
+// Copyright (C) The DDC development team, see COPYRIGHT.md file
+//
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <algorithm>
+#include <cassert>
+#include <cmath>
+#include <memory>
+
+#include "matrix.hpp"
+
+namespace ddc::detail {
+extern "C" int dgbtrf_(
+        int const* m,
+        int const* n,
+        int const* kl,
+        int const* ku,
+        double* a_b,
+        int const* lda_b,
+        int* ipiv,
+        int* info);
+extern "C" int dgbtrs_(
+        char const* trans,
+        int const* n,
+        int const* kl,
+        int const* ku,
+        int const* nrhs,
+        double* a_b,
+        int const* lda_b,
+        int* ipiv,
+        double* b,
+        int const* ldb,
+        int* info);
+
+template <class ExecSpace>
+class Matrix_Banded : public Matrix
+{
+protected:
+    int const m_kl; // no. of subdiagonals
+    int const m_ku; // no. of superdiagonals
+    int const m_c; // no. of columns in q
+    Kokkos::View<int*, Kokkos::HostSpace> m_ipiv; // pivot indices
+    // TODO: double**
+    Kokkos::View<double*, Kokkos::HostSpace> m_q; // banded matrix representation
```

Comment:
Is this TODO before merging?

Reply:
Looking at it again I think this is better to keep it 1D and follow closely the convention described in Lapack dgbtrs
```suggestion
    Kokkos::View<double*, Kokkos::HostSpace> m_q; // banded matrix representation
```